crates/bitwarden/src/crypto/enc_string.rs

```diff
@@ -395,33 +369,26 @@ fn invalid_len_error(expected: usize) -> impl Fn(Vec<u8>) -> EncStringParseError
 }
 
 impl LocateKey for EncString {}
-impl KeyEncryptable<EncString> for &[u8] {
-    fn encrypt_with_key(self, key: &SymmetricCryptoKey) -> Result<EncString> {
-        EncString::encrypt_aes256_hmac(self, key.mac_key.ok_or(CryptoError::InvalidMac)?, key.key)
+impl<Key: CryptoKey> KeyEncryptable<Key, EncString> for &[u8] {
+    fn encrypt_with_key(self, key: &Key) -> Result<EncString> {
+        key.encrypt(self)
     }
 }
 
-impl KeyDecryptable<Vec<u8>> for EncString {
-    fn decrypt_with_key(&self, key: &SymmetricCryptoKey) -> Result<Vec<u8>> {
-        match self {
-            EncString::AesCbc256_HmacSha256_B64 { iv, mac, data } => {
-                let mac_key = key.mac_key.ok_or(CryptoError::InvalidMac)?;
-                let dec = decrypt_aes256_hmac(iv, mac, data.clone(), mac_key, key.key)?;
-                Ok(dec)
-            }
-            _ => Err(CryptoError::InvalidKey.into()),
-        }
+impl<Key: CryptoKey> KeyDecryptable<Key, Vec<u8>> for EncString {
+    fn decrypt_with_key(&self, key: &Key) -> Result<Vec<u8>> {
+        key.decrypt(self)
```

Comment:
Does these interfaces still make sense? Why not simply call `key.decrypt` in the consumer.

Reply:
This is mostly a stopgap to get this working until I finish up another PR to get the EncString split into symm/asymm variants, then we can just have `impl KeyDecryptable<SymmCryptoKey> for EncString` and `impl KeyDecryptable<AsymmCryptoKey> for AsymmEncString`